### PR TITLE
feat: support project info

### LINKED ISSUE #22: Add project / project column properties to database pages
<sub>[Discuss in #**public-issue-22--add-project--project-column-properties-to-database-pages**](https://slack.com/app_redirect?team=TJ5G67VHU&channel=C02R9M4HTAT)</sub>

- What is the current functionality?

We don't show the project and project column, if any, this issue belongs to.

- What problems are you facing with the current feature?

Not knowing about the corresponding project

- What is your idea?

Adding two properties to the database to correspond to the project and project column.

Doing so requires the github token as the issue payload does not contain any project information, so we would have to query github for the information.

### PR DESCRIPTION
**Pull request by EndBug**
This PR adds support for two new text columns, named "Project" and "Project Column". Since the project info is not part of the issue event payload, we have to search through the repo projects until we find the corresponding card. We can optimize this by using the existing info in the table, if the issue has been updated.

It's not like it's really efficient, but I couldn't find a way to "search" a card directly based only on the repo without iterating through the different projects and columns.

Still to do:

- [ ] Update the Notion template
- [ ] ~Add tests?~

Fixes #22

### CODE REVIEW

**Comment by marissamarym** [comment]
(picking back up after the new year)

File: src/action.ts [hunk 1]
```diff
@@ -30,17 +44,65 @@ function parsePropertiesFromPayload(payload: IssuesEvent): CustomValueMap {
...
+): Promise<ProjectData | undefined> {
+  const {octokit, githubRepo, issueNumber, possible} = options;
+
+  const projects =
+    (
+      await octokit.rest.projects.listForRepo({
+        owner: githubRepo.split('/')[0],
+        repo: githubRepo.split('/')[1],
+      })
+    ).data || [];
```
Comment (nebularthm) [inline src/action.ts:75]: Does this get typed as `never`?

Comment (EndBug) [reply src/action.ts:75]: No, I see it all typed correctly... Where are you getting the error?

**Comment by EndBug** [comment]
I think that it's better to wait to add more tests since I've upgraded the API wrapper's version in #48, and would risk having to rewrite most of them because of the new API types

File: src/action.ts (hunk 1 above)
Comment (EndBug) [reply src/action.ts:76]: Marking this as resolved...

**Comment by EndBug** [comment]
Seems to be able to correctly set the project name and column 🎉 
image

**Comment by EndBug** [comment]
@marissamarym Yup, the template still needs to be updated by a member of Fabric on Notion

**Pull request merged**